src/test/java/htsjdk/utils/TestNGUtils.java

```diff
@@ -40,4 +42,41 @@
 
         return data.iterator();
     }
+
+    /**
+     * @return the cartesian product of two or more DataProviders than can be used as a new dataprovider
+     */
+    public static Object[][] cartesianProduct(Object[][] ... dataProviders){
```

Comment:
It is possible to write this and preserve types (e.g. avoid SuppressWarnings). I'm pretty sure it can also be written using generics (vs `Object`) but didn't have a chance to finish that.
```
    public static Object[][] cartesianProduct(Object[][]... dataProviders) {
        List<List<List<Object>>> lists = Arrays.stream(dataProviders)
                .map(Main::nestedArraysToNestedLists)
                .collect(Collectors.toList());
        final List<List<List<Object>>> product = Lists.cartesianProduct(lists);
        final List<List<Object>> mergeProduct = product.stream()
                .map(list -> {
                    List<Object> result = new ArrayList<>();
                    list.forEach(result::addAll);
                    return result;
                }).collect(Collectors.toList());
        return nestedListsToNestedArrays(mergeProduct);
    }

    /**
     * @param dataProvider a nested Object array
     * @return an equivalent nested List
     */
    public static<T> List<List<T>> nestedArraysToNestedLists(T[][] dataProvider) {
        return Arrays.stream(dataProvider)
                .map(Arrays::asList)
                .collect(Collectors.toList());
    }
    /**
     * @param lists a nested List
     * @return an equivalent nested array
     */
    public static Object[][] nestedListsToNestedArrays(List<List<Object>> lists) {
        return lists.stream().map(List::toArray).toArray(Object[][]::new);
    }
```

Reply:
Also I think `stream().map(... list.forEach().. return)` is really `flatMap`. something like
```
        final List<List<Object>> mergeProduct = product.stream().flatMap(list -> list.stream()).collect(Collectors.toList());
```

Reply:
* 👍 to the first comment, thats better. 
 * I don't think this can be done with generics in any sane way, since the It's an Object[][] of totally unrelated types.  There's no shared type other than Object.  Also, it's what dataproviders expect.
* That flatMap flattens the wrong level of list, this works but I'm not sure it's any clearer

```
  final List<List<List<Object>>> product = Lists.cartesianProduct(lists);
        final List<List<Object>> mergeProduct = product.stream()
                .map( l -> l.stream()
                        .flatMap(Collection::stream)
                        .collect(Collectors.toList()))
                .collect(Collectors.toList());
```

Reply:
Yeah, not sure much can be done here to make the code clearer. Regarding generics you can definitely make `cartesianProduct` generic. I ran into trouble with `nestedListsToNestedArrays` though. But as you say, the caller is always passing `Object` so not much is gained there.